tools/__tasks__/validate-head/index.js

```diff
@@ -1,5 +1,8 @@
 module.exports = {
     description: 'Validate commits',
-    task: [require('./javascript')],
+    task: [
+        require('./javascript'),
+        require('../validate/check-for-disallowed-strings'),
+    ],
```

Comment:
The intention of `validate-head` is to allow us to validate only changes that have been made in the current branch vs master. This would require a different `check-for-disallowed-strings` script with a bit more logic that runs your regex against [`changed files`](https://github.com/guardian/frontend/blob/master/tools/__tasks__/lib/get-changed-files.js). Let me know if you'd like any help with that

Reply:
The issue is that to be useful, this really needs to check all files (since it's looking for duplicates). I suppose I could add logic to check all files but only flag an error if some of the duplicates are in the changed files but it feels a bit overkill - really I just want to make use of the pre-push hook. Is there a better place to put this if that's all I need?

Reply:
Ah, I see your predicament. Would it be possible to grep for the disallowed strings in changed files and, if it finds a disallowed strings, grep for those strings against the rest of the application code? We ideally want to keep the pre-push hook as light as possible

Reply:
either that or leave it be caught in TC build? 

how likely is it someone will do it? is it something that will catch people out often or something we just don't want to ever happen and so needs catching, but is actually kind of unlikely?

Reply:
It's perhaps not that likely, so it's not the end of the world if we only do it in the TeamCity build.

I'll have a look optimising the pre-push grep in the way you suggested Simon. Though It's probably relevant that with no pathspec filters (so grepping the whole codebase) it only takes 0.654s to complete (though this is just for a single exclusion rule - in the worst case it would need to do this for each rule). With the `*.js *.scala *.html` pathspec it's 0.303s

Reply:
It's a fair point, perhaps my suggestion is a premature optimisation. 

I think if we wanted to impose the unimaginable 0.303 second delay upon everyone who pushes, I would probably be pedantic and suggest adding the call to `./tools/task-runner/runner validate/check-for-disallowed-strings` directly to the [`validate()`](https://github.com/guardian/frontend/blob/master/git-hooks/pre-push#L13) function of the `pre-push` hook.

`validate-head` should be for validation tasks that are checking for errors specifically introduced in this branch, rather than as a way of using the `pre-push` hook.

Reply:
I just tried your original suggestion @SiAdcock [here](https://github.com/guardian/frontend/compare/add-linter-for-disallowed-strings...disallowed-strings-validate-head?expand=1), only searching the whole codebase if there's a match in any of the changes files.

Funnily enough, it still took about 0.3 seconds to run, even when there were no changed files! Suggesting that the bulk of the running time of the command is taken up by bootstrapping common to all tasks, or by the `getChangedFiles()` task - either way, that particular optimisation is pointless in this case.

So I will go with your last suggestion, confident that I've tried everything! 😉